playground/nats/Nats.ApiService/Program.cs

```diff
@@ -0,0 +1,85 @@
+using NATS.Client.Core;
+using NATS.Client.JetStream;
+using NATS.Client.JetStream.Models;
+using Nats.Common;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+builder.AddServiceDefaults();
+
+builder.AddNats("nats", configureOptions: opts =>
+{
+    var jsonRegistry = new NatsJsonContextSerializerRegistry(AppJsonContext.Default);
+    return opts with { SerializerRegistry = jsonRegistry };
```

Comment:
Is this required or optional?

Reply:
it's optional. required for json serialization. Would you suggests to make it easier for ad hoc JSON serialization?

Reply:
I think so, but I don't know enough about your API design to suggest something.

Reply:
Challenge is our core NuGet packages are AOT ready and for ad-hoc JSON we have an extra package. I can bring NATS ad-hoc JSON package into Aspite.Nats.Client if AOT is not a concern (yet?).

Reply:
The Aspire component packages are intended to be AOT compatible if the associated client library is AOT compatible. For example, we are using the Configuration Binder source generator, so config binding works in native AOT.

We have 2 service apps in https://github.com/dotnet/eShop that support being published to native AOT:

* https://github.com/dotnet/eShop/blob/857fc5d18ea4e081b9e8a069763b037e3c695dec/src/Basket.API/Basket.API.csproj#L5
* https://github.com/dotnet/eShop/blob/857fc5d18ea4e081b9e8a069763b037e3c695dec/src/OrderProcessor/OrderProcessor.csproj#L6

So, yes, AOT is a concern for Aspire.

Reply:
Thanks for the comments. I'm going to leave this as it is for now since it looks like we're going to need some changes on NATS Client side as well (https://github.com/nats-io/nats.net.v2/issues/417)